server/src/main/java/org/elasticsearch/index/mapper/IndexFieldMapper.java

```diff
@@ -187,7 +175,7 @@ public Query wildcardQuery(String value,
 
         @Override
         public IndexFieldData.Builder fielddataBuilder(String fullyQualifiedIndexName) {
-            return new ConstantIndexFieldData.Builder(mapperService -> fullyQualifiedIndexName);
+            return new ConstantKeywordIndexFieldData.Builder(mapperService -> fullyQualifiedIndexName);
```

Comment:
No strong feelings, but I did it this way because `ConstantFieldType` makes no assumption about the type of the field, while `ConstantKeywordIndexFieldData` produces sorted-set doc values, which are specific to keywords (numerics would produce sorted-numeric doc values).